dropshot/src/error.rs

```diff
@@ -118,6 +118,17 @@ pub struct HttpError {
     pub external_message: String,
     /// Error message recorded in the log for this error
     pub internal_message: String,
+    /// Headers which should be added to error responses generated from this
+    /// error.
+    ///
+    /// The [`http::HeaderMap`] is boxed to reduce the size of `HttpError`s
+    /// which don't contain a `HeaderMap` (the common case).  Unlike containers
+    /// such as `Vec` and `HashMap`, an empty [`http::HeaderMap`] contains a
+    /// fairly large number of fields, rather than being a pointer to a heap
+    /// allocation containing the actual data.  Thus, we store an
+    /// `Option<Box<HeaderMap>>` here, so that the empty case is just a null
+    /// pointer.
```

Comment:
I would suggest replacing all of this with a `// Boxed in obeisance to clippy.`

Reply:
That's fair. I felt that including the justification for this in the API docs (rather than in an implementation comment) was worthwhile, since this _is_ exposed to consumers of the API, and they might wonder why this isn't just a `HeaderMap` when constructing/destructuring `HttpError`s. But, I'm happy to be convinced otherwise!

Reply:
I guess my perspective is that for the 99% of folks who don't care, it's a distraction. For the 1% who do, it's self-evident.

Reply:
Fair enough!